src/GslibIO.jl

```diff
@@ -52,28 +92,20 @@ Optionally set the value used for missings `na`.
 """
 function load_legacy(filename::AbstractString, dims::NTuple{3,Int};
                      origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
-  open(filename) do fs
-    # skip header
-    readline(fs)
 
-    # number of properties
-    nvars = parse.(Int, readline(fs))
+  # parse the GSLIB file
+  X, metadata = parse_gslib(filename)                     
 
-    # properties names
-    vars = [Symbol(readline(fs)) for i in 1:nvars]
+  vars = [Symbol(v) for v in metadata.varnames]
```

Comment:
I don't think this variable is needed. One can use `meta.varnames` directly?

Reply:
I need to check if the variable names need to be Symbols or just Strings.

Reply:
It seems line 99 (which is part the current code) only works with a array of Symbols. I could inject ```[Symbol(v) for v in metadata.varnames]``` in line 99 to avoid the vars variable, although in my view is clearer to use that variable.

Reply:
They need to be symbols. I would parse them as symbols inside the `parse_legacy` function.

Reply:
> They need to be symbols. I would parse them as symbols inside the `parse_legacy` function.

I prefer not to do it in the `parse_legacy` function, because the parser should be low level (variables names are just Strings). The use of Symbols is after parsing.

Reply:
Do we gain something from this string representation? I mean, are we planning to do any string processing? If not, please just convert the types to `Integer`, `String`, `Float64` etc. This will make the code clearer. If this was a major specification it would make more sense to save the raw strings, but here we only have a tiny header.